src/client/datascience/notebookStorage/baseModel.ts

```diff
@@ -115,14 +115,21 @@ export function updateNotebookMetadata(
         // Spec exists, just update name and display_name
         const name = kernelSpecOrModel.name || kernelSpecOrModel.display_name || '';
         const displayName = kernelSpecOrModel.display_name || kernelSpecOrModel.name || '';
+        const language = kernelSpecOrModel.language || kernelSpecOrModel.language || '';
+        const languageExistsInNBMetadata = 'language' in metadata.kernelspec;
```

Comment:
**Found a bug (unrelated change in PR - easy fix)**.
While testing this issue with a notebook containing java code found a bug.
* Open existing notebook (saved by Jupyter).
* Metadata will contain kernel spec information include language.
* We do not update the language in the kernel spec.
* Thus originally metadata in notebook pointed to java kernel with name & language as java.
* In VSC you choose Python & save it, 
Metadata has Python kernel name, but the metata for kernel language is still java.

Jupyter Notebooks always set the language, hence I'm adding it here as well.